src/transformers/models/perceiver/modeling_perceiver.py

```diff
@@ -2850,14 +2850,14 @@ def __init__(self, config: PerceiverConfig) -> None:
     def num_channels(self) -> int:
         return self.config.d_model
 
-    def forward(self, inputs: torch.LongTensor) -> torch.FloatTensor:
+    def forward(self, inputs: torch.LongTensor, pos: Optional[torch.Tensor] = None, network_input_is_1d: bool = True):
```

Comment:
Is there a benefit of having the `pos` and `network_input_is_1d` inputs here? It seems that these aren't used

Reply:
Good question! Yes, those parameters aren't used; they're really just to conform with the API call from the `PerceiverMultimodalPreprocessor` class [here](https://github.com/stevenmanton/transformers/blob/fix-perceiver-multimodal-bugs/src/transformers/models/perceiver/modeling_perceiver.py#L3392-L3394). It might be better to modify the [`AbstractPreprocssor`](https://github.com/stevenmanton/transformers/blob/fix-perceiver-multimodal-bugs/src/transformers/models/perceiver/modeling_perceiver.py#L2825) class to include this signature for the `forward`?